pkgs/applications/editors/vscode/extensions/ms-dotnettools-csharp/default.nix

```diff
@@ -1,65 +1,74 @@
 { lib
 , fetchurl
 , vscode-utils
-, unzip
 , patchelf
-, makeWrapper
 , icu
 , stdenv
 , openssl
-, mono
 }:
-
 let
-  # Get as close as possible as the `package.json` required version.
-  # This is what drives omnisharp.
-  rtDepsSrcsFromJson = lib.importJSON ./rt-deps-bin-srcs.json;
-
-  rtDepsBinSrcs = builtins.mapAttrs (k: v:
-      let
-        # E.g: "OmniSharp-x86_64-linux"
-        kSplit = builtins.split "(__)" k;
-        name = builtins.elemAt kSplit 0;
-        system = builtins.elemAt kSplit 2;
-      in
-      {
-        inherit name system;
-        installPath = v.installPath;
-        binaries = v.binaries;
-        bin-src = fetchurl {
-          urls = v.urls;
-          inherit (v) sha256;
-        };
-      }
-    )
-    rtDepsSrcsFromJson;
-
-  rtDepBinSrcByName = bSrcName:
-    rtDepsBinSrcs."${bSrcName}__${stdenv.targetPlatform.system}";
-
-  omnisharp = rtDepBinSrcByName "OmniSharp";
-  vsdbgs = [
-    (rtDepBinSrcByName "Debugger")
-  ] ++ lib.optionals (stdenv.isDarwin) [
-  # Include the aarch64-darwin debugger binaries on x86_64-darwin.  Even though OmniSharp will be
-  # running under Rosetta 2, debugging will fail to start if both sets of binaries are not present.
-    (rtDepsBinSrcs."Debugger__aarch64-darwin")
-  ];
-  razor = rtDepBinSrcByName "Razor";
+  vsixInfo = {
+    x86_64-linux = {
+      url = "https://github.com/OmniSharp/omnisharp-vscode/releases/download/v1.25.0/csharp-1.25.0-linux-x64.vsix";
+      sha256 = "1cqqjg8q6v56b19aabs9w1kxly457mpm0akbn5mis9nd1mrdmydl";
+      binaries = [
+        ".debugger/vsdbg-ui"
+        ".debugger/vsdbg"
+        ".omnisharp/1.39.0-net6.0/OmniSharp"
+        ".razor/createdump"
+        ".razor/rzls"
+      ];
+    };
+    aarch64-linux = {
+      url = "https://github.com/OmniSharp/omnisharp-vscode/releases/download/v1.25.0/csharp-1.25.0-linux-arm64.vsix";
+      sha256 = "0nsjgrb7y4w71w1gnrf50ifwbmjidi4vrw2fyfmch7lgjl8ilnhd";
+      binaries = [
+        ".debugger/vsdbg-ui"
+        ".debugger/vsdbg"
+        ".omnisharp/1.39.0-net6.0/OmniSharp"
```

Comment:
this should also be in one variable

Reply:
This version is the one bundled with the extension, not the one in nixpkgs

Reply:
yeah I know but we repeat that for every download. Would be nicer if we didn't.

Reply:
I mean, if we go that route, one could argue that every binary in this list could be in a variable then we combine them in the arrays

Reply:
However what I think I'll do is create sets of binaries and combine them based on which ones are available

Reply:
@SuperSandro2000 I've updated the PR with the approach I mentioned above, could you check it?